src/client/components/forms/profile.js

```diff
@@ -67,66 +87,168 @@ class ProfileForm extends React.Component {
 			});
 	}
 
+	valid() {
+		return this.birthDate.valid() && this.name.getValue();
+	}
+
 	render() {
 		const loadingElement =
 			this.state.waiting ? <LoadingSpinner/> : null;
-		const titles = this.props.titles.map((unlock) => {
-			const title = unlock.title;
-			title.unlockId = unlock.id;
-			return title;
-		});
 		const select2Options = {
 			width: '100%',
 			allowClear: true
 		};
+		const birthDate = this.state.birthDate;
+		const genderOptions = this.state.genders.map((gender) => ({
+			id: gender.id,
+			name: gender.name
+		}));
+		const titleOptions = this.state.titles.map((unlock) => {
+			const title = unlock.title;
+			title.unlockId = unlock.id;
+			return title;
+		});
+
+		const initialDisplayName = this.state.name;
+		const initialGender = this.state.gender ? this.state.gender.id : null;
+		const initialBio = this.state.bio;
+		const initialArea = injectDefaultAliasName(this.state.area);
+		let initialBirthDate = null;
+		if (birthDate) {
+			const year = birthDate.getFullYear();
+			const month = birthDate.getMonth() < 9 ?
+				`0${birthDate.getMonth() + 1}` : birthDate.getMonth() + 1;
+			const day = birthDate.getDay() < 10 ? `0${birthDate.getDay()}` :
+				birthDate.getDay();
+			initialBirthDate = `${year}-${month}-${day}`;
+		}
+
 		return (
-			<form
-				className="form-horizontal"
-				onSubmit={this.handleSubmit}
-			>
-				{loadingElement}
-				<Input
-					defaultValue={this.state.bio}
-					label="Bio"
-					labelClassName="col-md-3"
-					ref={(ref) => this.bio = ref}
-					type="textarea"
-					wrapperClassName="col-md-9"
-				/>
-				<Select
-					idAttribute="unlockId"
-					label="Title"
-					labelAttribute="title"
-					labelClassName="col-md-4"
-					options={titles}
-					placeholder="Select title"
-					ref={(ref) => this.title = ref}
-					wrapperClassName="col-md-4"
-				/>
-				<SearchSelect
-					noDefault
-					collection="area"
-					defaultValue={injectDefaultAliasName(this.state.area)}
-					label="Area"
-					labelClassName="col-md-4"
-					placeholder="Select area..."
-					ref={(ref) => this.area = ref}
-					select2Options={select2Options}
-					wrapperClassName="col-md-4"
-				/>
-				<div className="form-group">
-					<div className="col-md-4 col-md-offset-4">
-						<Button
-							block
-							bsSize="large"
-							bsStyle="primary"
-							type="submit"
+			<Grid>
+				<h1>Edit Profile</h1>
+				<Row>
+					<Col md={12}>
+						<p className="lead">Edit your public profile.</p>
+					</Col>
+				</Row>
+				<Row>
+					<Col
+						id="profileForm"
+						md={6}
+						mdOffset={3}
+					>
+						<form
+							className="form-horizontal"
+							onSubmit={this.handleSubmit}
 						>
-							Update!
-						</Button>
-					</div>
-				</div>
-			</form>
+							{loadingElement}
+							<Row>
```

Comment:
Hmm, let's remove the `<Row>` and `<Col>` elements surrounding the inputs, so that they take up the full width of the form. This may not look the best, but it'll be consistent with the entity editing PR, and then we can adjust both of them in the same ways to look good.